contracts/TokenDistributor.sol

```diff
@@ -0,0 +1,193 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {StakingRewardsV2} from "./StakingRewardsV2.sol";
+import {RewardEscrowV2} from "./RewardEscrowV2.sol";
+import {IKwenta} from "./interfaces/IKwenta.sol";
+import {ITokenDistributor} from "./interfaces/ITokenDistributor.sol";
+
+contract TokenDistributor is ITokenDistributor {
+    /// @notice represents the status of if a person already
+    /// claimed their epoch
+    mapping(address => mapping(uint => bool)) public claimedEpochs;
+
+    /// @notice kwenta interface
+    IKwenta public kwenta;
+
+    /// @notice rewards staking contract
+    StakingRewardsV2 public stakingRewardsV2;
+
+    /// @notice escrow contract which holds (and may stake) reward tokens
+    RewardEscrowV2 public rewardEscrowV2;
+
+    /// @notice last recorded balance of KWENTA in contract
+    uint public lastTokenBalance;
+
+    /// @notice last checkpoint time
+    uint public lastCheckpoint;
+
+    /// @notice starting week of deployment
+    uint public startTime;
+
+    /// @notice array for tokens allocated to each epoch
+    uint[1000000000000000] public tokensPerEpoch;
+
+    /// @notice the week offset in seconds
+    uint public offset;
+
+    constructor(
+        address _kwenta,
+        address _stakingRewardsV2,
+        address _rewardEscrowV2,
+        uint _offset
+    ) {
+        kwenta = IKwenta(_kwenta);
+        stakingRewardsV2 = StakingRewardsV2(_stakingRewardsV2);
+        rewardEscrowV2 = RewardEscrowV2(_rewardEscrowV2);
+
+        /// @notice custom start day (startTime + param)
+        if (_offset > 6) {
```

Comment:
It is slightly confusing that offset in the constructor is in days but elsewhere offset is in seconds. I'd recommend renaming the constructor param

Reply:
In fact, is it absolutely necessary to set the offset and many of these other related state variables in the constructor?

Reply:
6 should be a constant